### PR TITLE
Add split event processor for dataprepper

### LINKED ISSUE #6679: [DOC] Add docs for split event processor
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 New processor: [opensearch-project/data-prepper/pull/4166](https://github.com/opensearch-project/data-prepper/pull/4166)
This will be in Data Prepper 2.7

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by srikanthjg**
Closes #6679 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Naarcha-AWS** [comment]
@srikanthjg: Do you have a pipeline example with a response?

**Comment by Naarcha-AWS** [comment]
Won't be able to move forward with this one until I have usage example, similar to the one found here: https://opensearch.org/docs/latest/data-prepper/pipelines/configuration/processors/csv/#usage

**Comment by Naarcha-AWS** [comment]
@dlvenable: Can you take a look at this one?

File: _data-prepper/pipelines/configuration/processors/split-event.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+```
+{% include copy.html %}
+
+When an event contains the following input example:
+
+```
+{"query" : "open source", "some_other_field" : "abc" }
+```
+
+The input will be split into multiple events based on the field `query`, as the delimiter set as whitespace, as shown in the following example:
```
Comment (Naarcha-AWS) [inline _data-prepper/pipelines/configuration/processors/split-event.md:46]:
```suggestion
The input will be split into multiple events based on the field `query`, as the delimiter set as white space, as shown in the following example:
```

**Review by dlvenable — APPROVED** [review]

File: _data-prepper/pipelines/configuration/processors/split-event.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+nav_order: 56
+---
+
+# split-event
+
+The `split-event` processor is used to split events based on a delimiter and generates multiple events from a user-specified field.
+
+## Configuration
+
+The following table describes the configuration options for the `split-event`
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/split-event.md:15]:
```suggestion
The following table describes the configuration options for the `split-event` processor.
```

File: _data-prepper/pipelines/configuration/processors/split-event.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+| Option           | Type    | Description                                                                                   |
+|------------------|---------|-----------------------------------------------------------------------------------------------|
+| `field`          | String  | The field in the event to be split.                                                           |
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/split-event.md:19]:
```suggestion
| `field`          | String  | The event field to be split.                                                           |
```

File: _data-prepper/pipelines/configuration/processors/split-event.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  processor:
+    - split_event:
+        field: query
+        delimiter: ' '    
+  sink:
+    - stdout:
+```
+{% include copy.html %}
+
+When an event contains the following input example:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/split-event.md:40]:
```suggestion
When an event contains the following example input:
```

File: _data-prepper/pipelines/configuration/processors/split-event.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+```
+{"query" : "open source", "some_other_field" : "abc" }
+```
+
+The input will be split into multiple events based on the field `query`, as the delimiter set as white space, as shown in the following example:
```
Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/split-event.md:46]:
```suggestion
The input will be split into multiple events based on the `query` field, as the delimiter set as white space, as shown in the following example:
```

Comment (natebower) [inline _data-prepper/pipelines/configuration/processors/split-event.md:46]: "with" the delimiter set to white space (instead of "as")?

**Review by natebower — COMMENTED** [review]
@srikanthjg @Naarcha-AWS Just a few comments/changes. Thanks!

**Pull request merged**